onnxruntime/contrib_ops/cpu/bert/attention.cc

```diff
@@ -171,11 +182,74 @@ Status Attention<T>::PrePack(const Tensor& weights, int input_idx, AllocatorPtr
     return Status::OK();
   }
 
+  if (save_prepacked_initializers) {
+    ConvertPrePackWeightsIntoTensor(alloc, weights, prepacked_weights);
+  }
+
   is_packed = true;
   is_prepack_ = true;
   return Status::OK();
 }
 
+template <typename T>
+void Attention<T>::ConvertPrePackWeightsIntoTensor(onnxruntime::AllocatorPtr& alloc,
+                                                   const onnxruntime::Tensor& weights,
+                                                   PrePackedWeights* prepacked_weights) {
+  // buffer of packed_tensor is combine of:
+  // 1. packed_weights: packed_weights_[0], packed_weights_[1], packed_weights_[2]
+  // 2. weight shape: first vector memory size, then vector content
+  // 3. original packed_weights buffer in order
+
+  TensorShapeVector shape_vector = weight_shape_.AsShapeVector();
+  size_t shape_vector_mem_size = utils::CalculateTensorShapeVectorMemoryUsage(shape_vector);
+  void* shape_vector_ptr = static_cast<void*>(&shape_vector);
```

Comment:
This is undefined behavior. You need to copy data not the object of this class.
For this you can just get GetDims() span and using it copy data from it.